### PR TITLE
Affine_Substitution_Cipher

### PR DESCRIPTION
**Pull request by nutanaarohi123**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

#486 
The Affine cipher is a type of monoalphabetic substitution cipher, wherein each letter in an alphabet is mapped to its numeric equivalent, encrypted using a simple mathematical function, and converted back to a letter. The output is attached below:

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#486 )

Screenshot (74)

### CODE REVIEW

**Review by SKAUL05 — CHANGES_REQUESTED** [review]
@nutanaarohi123 Include Space and Time Complexity at the end

File: C/cryptography/Affine_Substitution_Cipher.c [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+#include<string.h>
+#include<ctype.h>
+#include<stdlib.h>
+// function declaration
+int CalculateGCD(int);
+int CalculateMultiplicativeInverse(int);
+
+int main() {
+  int index, indx, k, gcd, alpha, beta, num_msg[100], numcipher[100], numdecipher[100], decipher[100], alphaInverse;
+  char msg[100], cipher[100];
```
Comment (vatsalkesarwani12) [inline C/cryptography/Affine_Substitution_Cipher.c]: Don't restrict your cipher for 100 lengths make it more scalable.

Take the size dynamically

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

**Comment by nutanaarohi123** [comment]
> @nutanaarohi123 Include Space and Time Complexity at the end
Updated required changes. Kindly review @SKAUL05

File: C/cryptography/Affine_Substitution_Cipher.c (hunk 1 above)
Comment (nutanaarohi123) [reply C/cryptography/Affine_Substitution_Cipher.c]: Updated the file as required. Kindly review @plazzy99

**Review by SKAUL05 — APPROVED** [review]

**Comment by nutanaarohi123** [comment]
Kindly review @atarax665 @iamrajiv @plazzy99 .

File: C/cryptography/Affine_Substitution_Cipher.c [hunk 2]
```diff
@@ -0,0 +1,158 @@
+// C program to illustate Affine Cipher
```
Comment (iamrajiv) [inline C/cryptography/Affine_Substitution_Cipher.c]: @nutanaarohi123 add little bit description about Affine Substitution Cipher

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: C/cryptography/Affine_Substitution_Cipher.c (hunk 2 above)
Comment (nutanaarohi123) [reply C/cryptography/Affine_Substitution_Cipher.c]: > @nutanaarohi123 add little bit description about Affine Substitution Cipher

Added .Kindly review @iamrajiv

**Review by iamrajiv — APPROVED** [review]
@nutanaarohi123 LGTM!

**Comment by nutanaarohi123** [comment]
@plazzy99  @atarax665 Kindly review this..been three days already.

**Review by vatsalkesarwani12 — APPROVED** [review]
LGTM 👍

**Pull request merged**